src/covid19model/models/models.py

```diff
@@ -149,36 +152,45 @@ def integrate(t, S, E, I, A, M, ER, C, C_icurec, ICU, R, D, H_in, H_out, H_tot,
-        vacc_eligible = S + R + E + I + A
+        # vaccination eligible states
+        VE = S + R + E + I + A
 
         # Compute infection pressure (IP) of both variants
         # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-        IP_old = (1-alpha)*beta*s*np.matmul(Nc,((I+A)/T))
-        IP_new = alpha*K*beta*s*np.matmul(Nc,((I+A)/T))
+        if Nc is None:
+            print(t)
+        IP_old = (1-alpha)*beta*s*np.matmul(Nc,((I+A+leakiness*V)/T)) # leakiness
+        IP_new = alpha*K*beta*s*np.matmul(Nc,((I+A+leakiness*V)/T))
 
         # Compute the  rates of change in every population compartment
         # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-        dS  = - (IP_old + IP_new)*S + zeta*R - v*e*S - N_vacc/vacc_eligible*S
-        dE  = (IP_old + IP_new)*S - E/sigma - v*e*E - N_vacc/vacc_eligible*E + (IP_old + IP_new)*(1-e)*V 
-        dI = (1/sigma)*E - (1/omega)*I - N_vacc/vacc_eligible*I
-        dA = (a/omega)*I - A/da - N_vacc/vacc_eligible*A        
+        dS  = - (IP_old + IP_new)*S + zeta*R - v*e*S - N_vacc/VE*S
+        dE  = (IP_old + IP_new)*S - E/sigma - v*e*E - N_vacc/VE*E + (IP_old + IP_new)*(1-e)*V 
+        dI = (1/sigma)*E - (1/omega)*I - N_vacc/VE*I
+        dA = (a/omega)*I - A/da - N_vacc/VE*A        
         dM = ((1-a)/omega)*I - M*((1-h)/dm) - M*h/dhospital
         dER = M*(h/dhospital) - (1/der)*ER
         dC = c*(1/der)*ER - (1-m_C)*C*(1/dc_R) - m_C*C*(1/dc_D)
         dC_icurec = ((1-m_ICU)/dICU_R)*ICU - C_icurec*(1/dICUrec)
         dICUstar = (1-c)*(1/der)*ER - (1-m_ICU)*ICU/dICU_R - m_ICU*ICU/dICU_D
-        dR  = A/da + ((1-h)/dm)*M + (1-m_C)*C*(1/dc_R) + C_icurec*(1/dICUrec) - zeta*R +  v*e*S + v*e*E - N_vacc/vacc_eligible*R
+        dR  = A/da + ((1-h)/dm)*M + (1-m_C)*C*(1/dc_R) + C_icurec*(1/dICUrec) - zeta*R +  v*e*S + v*e*E - N_vacc/VE*R
         dD  = (m_ICU/dICU_D)*ICU + (m_C/dc_D)*C
         dH_in = M*(h/dhospital) - H_in
         dH_out =  (1-m_C)*C*(1/dc_R) +  m_C*C*(1/dc_D) + (m_ICU/dICU_D)*ICU + C_icurec*(1/dICUrec) - H_out
         dH_tot = M*(h/dhospital) - (1-m_C)*C*(1/dc_R) -  m_C*C*(1/dc_D) - (m_ICU/dICU_D)*ICU - C_icurec*(1/dICUrec)
-        dV_new = N_vacc/vacc_eligible*S + N_vacc/vacc_eligible*R + N_vacc/vacc_eligible*E + N_vacc/vacc_eligible*I + N_vacc/vacc_eligible*A - V_new
-        dV = N_vacc/vacc_eligible*S + N_vacc/vacc_eligible*R + N_vacc/vacc_eligible*E + N_vacc/vacc_eligible*I + N_vacc/vacc_eligible*A - (IP_old + IP_new)*(1-e)*V
-        # If A and I are both zero, a division error occurs
+        dV_new = N_vacc/VE*S + N_vacc/VE*R + N_vacc/VE*E + N_vacc/VE*I + N_vacc/VE*A - V_new
+        dV = N_vacc/VE*S + N_vacc/VE*R + N_vacc/VE*E + N_vacc/VE*I + N_vacc/VE*A - (IP_old + IP_new)*(1-e)*V
+        dVE = dS + dR + dE + dI + dA
         # Update fraction of new COVID-19 variant
         dalpha = IP_new/(IP_old+IP_new) - alpha
+         # If A and I are both zero, a division error occurs
+        dalpha[np.isnan(dalpha)] = 0
```

Comment:
When using one exposed individual as the initial condition, both `IP_old` and `IP_new` are zero, doesn't the algorithm then 'get stuck' while trying to compute dalpha? I.e. does this protection work in practice? If not, maybe a `try`, `except` structure can help?

Reply:
mmh... good question... something to keep in mind